st2client/st2client/commands/action_alias.py

```diff
@@ -40,3 +49,98 @@ class ActionAliasGetCommand(resource.ContentPackResourceGetCommand):
+                                       'output. "all" will return all '
+                                       'attributes.'))
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        match, _ = self.manager.match(args.match_text, **kwargs)
+        return [match]
+
+    def run_and_print(self, args, **kwargs):
+        instances = self.run(args, **kwargs)
+        self.print_output(instances, table.MultiColumnTable,
+                          attributes=args.attr, widths=args.width,
+                          json=args.json, yaml=args.yaml)
+
+
+class ActionAliasExecuteCommand(resource.ResourceCommand):
+    display_attributes = ['name']
+
+    def __init__(self, resource, *args, **kwargs):
+        super(ActionAliasExecuteCommand, self).__init__(
+            resource, 'execute',
+            'Execute the command text by finding a matching ActionAlias.',
+            *args, **kwargs)
+
+        self.parser.add_argument('command_text',
+                                 metavar='command',
+                                 help=help)
+        self.parser.add_argument('-h', '--help',
+                                 action='store_true', dest='help',
+                                 help='Print usage for the given action.')
+        self.parser.add_argument('--trace-tag', '--trace_tag',
+                                 help='A trace tag string to track execution later.',
+                                 dest='trace_tag', required=False)
+        self.parser.add_argument('--trace-id',
+                                 help='Existing trace id for this execution.',
+                                 dest='trace_id', required=False)
+        self.parser.add_argument('-a', '--async',
+                                 action='store_true', dest='async',
+                                 help='Do not wait for action to finish.')
+        self.parser.add_argument('-u', '--user', type=str, default=None,
+                                 help='User under which to run the action (admins only).')
+
+        self.parser.add_argument('--attr', nargs='+',
+                                 default=self.display_attributes,
+                                 help=('List of attributes to include in the '
+                                       'output. "all" will return all '
+                                       'attributes.'))
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        action_alias, representation = self.manager.match(args.command_text, **kwargs)
```

Comment:
the patterns got a bit wonky crossing resource types between controllers. I'll have a think about it.